apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -1161,10 +1155,65 @@ handle_info({force_update_presence, LUser}, StateName,
             StateData
     end,
     {next_state, StateName, NewStateData};
-handle_info({send_filtered, Feature, From, To, Packet}, StateName, StateData) ->
+handle_info(resume_timeout, resume_session, StateData) ->
+    {stop, normal, StateData};
+handle_info(check_buffer_full, StateName, StateData) ->
+    case is_buffer_full(StateData#state.stream_mgmt_buffer_size,
+                        StateData#state.stream_mgmt_buffer_max) of
+        true ->
+            Err = ?RESOURCE_CONSTRAINT_ERR((StateData#state.lang),
+                                           <<"too many unacked stanzas">>),
+            send_element(StateData, Err),
+            send_trailer(StateData),
+            {stop, normal, StateData};
+        false ->
+            fsm_next_state(StateName,
+                           StateData#state{stream_mgmt_constraint_check_tref = undefined})
+    end;
+handle_info({store_session_info, User, Server, Resource, KV, _FromPid}, StateName, StateData) ->
+    ejabberd_sm:store_info(User, Server, Resource, KV),
+    fsm_next_state(StateName, StateData);
+handle_info(Info, StateName, StateData) ->
+    handle_incoming_message(Info, StateName, StateData).
+
+maybe_terminate(Acc) ->
```

Comment:
That's fine it's transitional, still I think it'd be better to change the function name.